strawberryfields/ops.py

```diff
@@ -1852,31 +1839,43 @@ def _decompose(self, reg):
 
         if self.pure and is_diag:
             # covariance matrix consists of x/p quadrature squeezed state
-            for n, expr in enumerate(D[:self.ns]*2/self.hbar):
-                if np.abs(expr - 1) >= _decomposition_merge_tol:
+            for n, expr in enumerate(D[:self.ns]):
+                if np.abs(expr - 1) >= _decomposition_tol:
                     r = abs(log(expr)/2)
-                    cmds.append(Command(Sgate(r, 0), reg[n]))
+                    cmds.append(Command(Squeezed(r, 0), reg[n]))
+                else:
+                    cmds.append(Command(Vac, reg[n]))
```

Comment:
This catches the case where `r=0`, correct? What was happening before?

Reply:
Previously, it would simply run `Sgate(0)`, which was incorrect. Indicates that even though we have 100% test coverage here, our test suite was not sufficient as this was not flagged.

Reply:
Previously when r=0 it would do nothing, add no gates at all.

Reply:
True :thinking: My comment was incorrect.